src/main/steps/common/yourhearings/hearings/content.ts

```diff
@@ -59,8 +143,164 @@ export const form: FormContent = {
 
 export const generateContent: TranslationFn = content => {
   const translations = languages[content.language]();
+  updateCaseDataForHearings(content.additionalData?.req);
   return {
     ...translations,
     form: { ...form, fields: (form.fields as FormFieldsFn)(content.userCase || {}) },
   };
 };
+
+const updateCaseDataForHearings = (req: AppRequest<Partial<Case>>): void => {
```

Comment:
change the method name to prepareHearingData and return the hearing list as needed in template.njk

Reply:
Do not store anything in session

Reply:
Yes modified all the logic